apps/wallet/src/ui/app/pages/home/nfts/HiddenAsset.tsx

```diff
@@ -65,6 +66,28 @@ export default function HiddenAsset(item: HiddenAssetProps) {
             collectibleType: type!,
         });
     }
+
+    async function handleShowAsset() {
+        try {
+            await showAsset(objectId);
+            toast.success(
+                (t) => (
+                    <MovedAssetNotification
+                        t={t}
+                        destination="Visual Assets"
+                        onUndo={() => undoShowAsset(objectId)}
```

Comment:
In the other hand, this can actually throw an error

If the reason you made an `undo` version of both `hide` and `show` assets is because one is suppose to have an internal try-catch and the other not, I would remove both `undo` methods and let the two `hide` and `show` methods throw an error. Then you decide whether to use `try-catch` or not in the UI, not in the impl itself